### DISCUSSION TITLE
Allow partial success during parsing

### DISCUSSION THREAD

**Discussion by xhw994** [colinhacks/zod#945] [Q&A] [answered]
```ts
const zNullableString = z
  .string()
  .nullish()
  .transform((s) => s ?? '');

const zApiResponse = z.object({
  total: z.number(),
  pagination: z.object({
    from: z.number().int().positive(),
    size: z.number().int().positive(),
  }),
  records: z.array(zRecord),
});

const zRecord = z.object({
  id: z.string().uuid(),
  name: zNullableString,
  email: zNullableString,
  address: z
    .object({
      street: zNullableString,
      city: zNullableString,
      state: zNullableString,
      country: zNullableString,
    })
    .nullable(),
});
```

Say I have the above schema. The server can return multiple `zRecord`. The problem is when one of the record failed to parse, it will abort and throw an error. What I want instead is to skip the failed record and continue parsing. And after parsing all records, I get a `data` object containing the parsed API response, and an `error` object containing the Zod errors. Is this possible?

**Comment by JacobWeisenburger** [comment, marked as answer]
Unfortunately, I don't think there is a way to do that at this time. I asked a similar question awhile back: https://github.com/colinhacks/zod/discussions/875
